h2o-algos/src/main/java/hex/tree/DHistogram.java

```diff
@@ -192,33 +194,33 @@ public DHistogram(String name, final int nbins, int nbins_cats, byte isInt, doub
     }
     _nbin = (char) xbins;
     assert(_nbin>0);
-    assert(_w ==null);
-    assert(_wY ==null);
-    assert(_wYY ==null);
+    assert(_vals ==null);
+
 //    Log.info("Histogram: " + this);
     // Do not allocate the big arrays here; wait for scoreCols to pick which cols will be used.
   }
 
   // Interpolate d to find bin#
   public int bin( double col_data ) {
-    assert( !Double.isNaN(col_data) ); //NAs go to a separate bucket
+    if(Double.isNaN(col_data)) return _nbin; // NA bucket
```

Comment:
Is it clear why this condition is now possible?  I thought the logic of the histogram content didn't change in this PR.